src/main/java/quizapp/app/favorite/FavoriteController.java

```diff
@@ -0,0 +1,64 @@
+package quizapp.app.favorite;
+
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.security.core.annotation.AuthenticationPrincipal;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.validation.BindingResult;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ModelAttribute;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import quizapp.domain.model.Quiz;
+import quizapp.domain.model.User;
+import quizapp.domain.repository.quiz.QuizRepository;
+import quizapp.domain.repository.user.UserRepository;
+import quizapp.domain.service.quiz.QuizService;
+import quizapp.domain.service.user.UserDetailsImpl;
+
+@Controller
+@RequestMapping("quiz/favorites")
+public class FavoriteController {
+  @Autowired
+  QuizService quizService;
+  @Autowired
+  UserRepository userRepository;
+  @Autowired
+  QuizRepository quizRepository;
+
+  @ModelAttribute
+  FavoriteForm setUpForm() {
+    FavoriteForm form = new FavoriteForm();
+    return form;
+  }
+
+  @GetMapping
+  String favorites(Model model, @AuthenticationPrincipal UserDetailsImpl userDetailsImpl) {
+    List<Quiz> favorites = userDetailsImpl.getUser().getFavoriteQuizList();
+    model.addAttribute("favorites", favorites);
+    return "quiz/favorites";
+  }
+
+  @PostMapping
+  String favorite(@ModelAttribute FavoriteForm form, BindingResult bindingResult,
+      @AuthenticationPrincipal UserDetailsImpl userDetailsImpl, Model model) {
+    if (bindingResult.hasErrors()) {
+      return favorites(model, userDetailsImpl);
+    }
+    User currentUser = userDetailsImpl.getUser();
+    Optional<Quiz> opt = quizRepository.findById(form.getQuizId());
+    Quiz quiz = opt.get();
+
+    currentUser.getFavoriteQuizList().add(quiz);
```

Comment:
getFavoriteQuizList().add(quiz)というのは、日本語でいうと「クイズをお気に入りに登録する」という処理をしているのだと思います。それをUserクラスのメソッドとして表現した方がDDDっぽいというか、BOSっぽいコードになります。

例えばですが、UserクラスにfavorateQuiz()というメソッドを用意して、ここでは
```suggestion
    user.favorateQuiz(quiz);
```
として、
Userクラス内で
```
public void favorateQuiz(quiz) {
  favoriteQuizList.add(quiz);
}
```
みたいにしておくと、良いかなと思います。

あと細かいですが、current と付いていると状態がどんどん変化していくものをイメージしましたので、currentではなくて単にuserくらいの方がいいかもです。
例）
userのcuurentStatus（ステータスはどんどん変わるが、現在のステータスを明示している）
for (File currentFile : fileList) {}（たくさんのファイルが処理対象になっていて、現在のユーザーを明示している）

Reply:
ありがとうございます、修正いたします。
rubyではログイン中のユーザーのことをcurrent_userとして扱っていたのでそう言うものだと思っていました。そちらも修正します！

Reply:
なるほどですね、ログイン中のという意味をあまり考えていませんでした。
ついていてもそこまで不自然でもないと思いますので、そのままでもOKです。

Reply:
かしこまりました。個人的に分かりやすいという理由でそのままでいこうと思います。